### PR TITLE
Convert several data classes to Kotlin.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Fixes #

**Description of the Change**
Convert several classes primarily used to store data (mostly those used for RPC calls) to Kotlin. Since Kotlin automatically generates a lot of the boilerplate code associated with such classes, this allows said classes to be more concise without impacting functionality.

**Release Notes**
N/A

### CODE REVIEW

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Transfer.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of BOINC.
+ * http://boinc.berkeley.edu
+ * Copyright (C) 2012 University of California
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/Transfer.kt:4]: Fix the year, please

File: android/BOINC/app/src/test/java/edu/berkeley/boinc/rpc/WorkUnitsParserTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        workUnitsParser.characters("1.5".toCharArray(), 0, 3);
+        workUnitsParser.endElement(null, WorkUnit.Fields.RSC_DISK_BOUND, null);
+        workUnitsParser.endElement(null, WorkUnitsParser.WORKUNIT_TAG, null);
+
+        expected.setName(WORK_UNIT_NAME);
+        expected.setRscDiskBound(1.5);
+
+        assertEquals(Collections.singletonList(expected), workUnitsParser.getWorkUnits());
+    }
+}
```
Comment (AenBleidd) [inline android/BOINC/app/src/test/java/edu/berkeley/boinc/rpc/WorkUnitsParserTest.java:238]: Please add newline at the end of file

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/PrefsListItemWrapperClasses.kt [hunk 3]
```diff
@@ -0,0 +1,140 @@
+/*
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/PrefsListItemWrapperClasses.kt:1]: Please move classes in this file into separate one.
Generic rule is 1 class - 1 file except. We're not running out of empty space to store list of files so there is no need to put several classes into one file. Also it's easier to search for a class just looking at the name of the file.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/AccountManagerClasses.kt [hunk 4]
```diff
@@ -0,0 +1,117 @@
+/*
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/AccountManagerClasses.kt:1]: Please split this file too (see my previous comment)

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/ProjectRelatedClasses.kt [hunk 5]
```diff
@@ -0,0 +1,163 @@
+/*
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/ProjectRelatedClasses.kt:1]: Please split this file too (see my previous comment)

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/PrefsListItemWrapperClasses.kt (hunk 3 above)
Comment (Isira-Seneviratne) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/PrefsListItemWrapperClasses.kt:1]: In Kotlin, it's encouraged to place multiple related declarations in the same file: https://kotlinlang.org/docs/reference/coding-conventions.html#source-file-organization

**Pull request merged**